src/puppetlabs/puppetdb/jdbc.clj

```diff
@@ -431,7 +436,8 @@
      (doto config
        (.setJdbcUrl (str "jdbc:" subprotocol ":" subname))
        (.setAutoCommit false)
-       (.setInitializationFailFast false))
+       (.setInitializationFailFast false)
+       (.setTransactionIsolation "TRANSACTION_READ_COMMITTED"))
```

Comment:
Huh, turns out it is. I don't feel too bad about setting it here though. The real fix is to not re-set it on every query.